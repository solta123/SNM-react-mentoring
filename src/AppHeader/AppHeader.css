.root {
  flex-grow: 1;
}

.title {
  flex-grow: 1;
  text-align: left;
}

.AddMovieModal {
  max-width: 400px;
  max-height: 800px;
  padding-bottom: 2rem;
  margin: auto;
}

.AddMovieForm > div {
  margin: 0.5rem 2rem;
}

.AddMovieForm > div > div {
  width: 100%;
}

.AddMovieFormButtons {
  float: right;
  margin-left: 0.3rem !important;
}

.ModalContainer {
  padding: 1rem 1rem 3rem 1rem;
}

.closeIcon {
  position: absolute;
  right: 1rem;
  top: 1.3rem;
  cursor: pointer;
}
