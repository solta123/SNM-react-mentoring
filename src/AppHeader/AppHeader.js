import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    title: {
        flexGrow: 1,
        textAlign: 'left'
    },
    addIcon: {
        paddingRight: '8px'
    }
}));

const AppHeader = () => {
    const classes = useStyles();
  
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
              <Typography variant="h6" className={classes.title}>
                  <b>netflix</b>Roulette
              </Typography>
              <Button color="inherit" variant="outlined">
                  <AddIcon></AddIcon>Add movie
              </Button>
          </Toolbar>
        </AppBar>
      </div>
    );
}

export default AppHeader;
