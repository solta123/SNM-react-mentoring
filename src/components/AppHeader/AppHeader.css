.root {
  flex-grow: 1;
}

.title {
  flex-grow: 1;
  text-align: left;
  color: white;
  text-decoration: none;
}

.AddMovieModal {
  max-width: 400px;
  max-height: 800px;
  margin: auto;
}

.MuiDialogContent-root {
  padding: 0 !important;
}
