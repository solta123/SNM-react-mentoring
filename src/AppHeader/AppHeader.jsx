import './AppHeader.css';
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import AddMovieModal from '../AddMovieModal/AddMovieModal';
import Modal from '@material-ui/core/Modal';
import { DialogContent } from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import { initialState } from '../store/reducers/movie';
import { getMovies } from '../store/actionCreator';
import { connect } from 'react-redux';
import * as actionTypes from '../store/actions';

const AppHeader = props => {
  return (
    <div className="root">
      <AppBar position="static">
        <Toolbar>
          <NavLink to="/" className="title" onClick={props.getMovies}>
            <Typography variant="h6">
              <b>netflix</b>Roulette
            </Typography>
          </NavLink>
          <Button color="inherit" variant="outlined" onClick={() => props.handleModal(true)}>
            <AddIcon></AddIcon>Add movie
          </Button>
          <Modal open={props.isAddModalOpen} className="AddMovieModal" onClose={() => props.handleModal(false)}>
            <DialogContent>
              <AddMovieModal onCloseModal={() => props.handleModal(false)} />
            </DialogContent>
          </Modal>
        </Toolbar>
      </AppBar>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    isAddModalOpen: state.movie.isAddModalOpen
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getMovies: () => dispatch(getMovies({ ...initialState, search: '' })),
    handleModal: value => dispatch({ type: actionTypes.MODAL, value: value }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppHeader);
