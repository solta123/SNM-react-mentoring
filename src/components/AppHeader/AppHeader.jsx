import './AppHeader.css';
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import AddMovieModal from '../AddMovieModal/AddMovieModal';
import Modal from '@material-ui/core/Modal';
import { DialogContent, Select, MenuItem, FormControl } from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import { setToDefaultState } from '../store/actionCreator';
import { connect } from 'react-redux';
import * as actionTypes from '../store/actions';
import PropTypes from 'prop-types';
import { useTranslation } from "react-i18next";

const AppHeader = props => {
    const { t, i18n } = useTranslation('common');
    const tSource = 'NETFLIXROULETTE.APP_HEADER.';

  const handleLangChange = event => {
    i18n.changeLanguage(event.target.value);
    props.handleLang(event.target.value);
  };

  return (
    <div className="root">
      <AppBar position="static">
        <Toolbar>
          <NavLink to="/" className="title" onClick={props.getMovies}>
            <Typography variant="h6">
              <b>netflix</b>Roulette
            </Typography>
          </NavLink>
          <Button id="AddMovieButton" color="inherit" variant="outlined" onClick={() => props.handleModal(true)}>
            <AddIcon></AddIcon>{t(tSource + 'ADD_MOVIE')}
          </Button>
          <FormControl id="lang-button">
            <Select id="lang-select" value={props.lang} onChange={handleLangChange}>
              <MenuItem id="en" value="en">en</MenuItem>
              <MenuItem id="hu" value="hu">hu</MenuItem>
            </Select>
          </FormControl>
          <Modal open={props.isAddModalOpen} className="AddMovieModal">
            <DialogContent>
              <AddMovieModal onCloseModal={() => props.handleModal(false)} />
            </DialogContent>
          </Modal>
        </Toolbar>
      </AppBar>
    </div>
  );
}

AppHeader.propTypes = {
  isAddModalOpen: PropTypes.bool,
  handleModal: PropTypes.func,
  getMovies: PropTypes.func,
  lang: PropTypes.string,
  handleLang: PropTypes.func
}

const mapStateToProps = state => {
  return {
    isAddModalOpen: state.movie.isAddModalOpen,
    lang: state.movie.lang
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getMovies: () => dispatch(setToDefaultState()),
    handleModal: value => dispatch({ type: actionTypes.MODAL, value }),
    handleLang: lang => dispatch({ type: actionTypes.LANG, lang })
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppHeader);
