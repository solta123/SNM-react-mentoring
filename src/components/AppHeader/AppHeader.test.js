import React, { Suspense } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import AppHeader from './AppHeader';
import { I18nextProvider } from "react-i18next";
import i18next from "i18next";
import { createStore, combineReducers } from 'redux';
import { mount } from "enzyme";
import movieReducer from '../store/reducers/movie';
import common_hu from "../../translations/hu/common.json";
import common_en from "../../translations/en/common.json";
import { act } from 'react-dom/test-utils';

describe('AppHeader', () => {
    let wrapper;
    let store;

    beforeEach(() => {
        store = createStore(
            combineReducers({
                movie: movieReducer
            })
        );

        i18next.init({
            interpolation: { escapeValue: false },
            lng: 'en',
            resources: {
                en: {
                    common: common_en
                },
                hu: {
                    common: common_hu
                },
            },
        });

        act(() => {
            wrapper = mount(
                <Suspense fallback="loading">
                    <I18nextProvider i18n={i18next}>
                        <MemoryRouter>
                            <Provider store={store}>
                                <AppHeader />
                            </Provider>
                        </MemoryRouter>
                    </I18nextProvider>
                </Suspense>
            );
        });
    });

    it('should match snapshot', () => {
        const tree = renderer.create(
            <Suspense fallback="loading">
                <I18nextProvider i18n={i18next}>
                    <MemoryRouter>
                        <Provider store={store}>
                            <AppHeader />
                        </Provider>
                    </MemoryRouter>
                </I18nextProvider>
            </Suspense>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should open add movie modal', () => {
        act(() => {
            wrapper.find('button#AddMovieButton').simulate('click');
        });
        expect(store.getState().movie.isAddModalOpen).toEqual(true);
        expect(store.getState().movie.editableMovie).toBeFalsy();
    });

    // it('should change language', () => {
    //     wrapper.find('div#lang-select').simulate('click');
    //     wrapper.find('li#hu').simulate('click');
    //     expect(store.getState().movie.lang).toEqual('hu');
    // });
});