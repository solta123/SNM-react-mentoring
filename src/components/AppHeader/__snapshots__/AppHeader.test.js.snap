// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="root"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        aria-current="page"
        class="title active"
        href="/"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          <b>
            netflix
          </b>
          Roulette
        </h6>
      </a>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add movie
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
